src/onnxruntime.cc

```diff
@@ -1236,6 +1270,15 @@ ModelInstanceState::SetInputTensors(
   RESPOND_ALL_AND_RETURN_IF_ERROR(
       responses, request_count,
       TRITONBACKEND_RequestInputCount(requests[0], &input_count));
+  std::vector<std::pair<TRITONSERVER_MemoryType, int64_t>> allowed_input_types;
```

Comment:
This allowed_input_types should only be used by the first case below so should move inside conditional to where it was. For the non-GPU case you still want to have CPU_PINNED and CPU for the case where the data is coming from GPU and so will be copied to CPU_PINNED.